src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/model/ContextObject.java

```diff
@@ -446,7 +446,7 @@ public void setReceiver(final Object value) {
 
     @TruffleBoundary
     public Object atTemp(final int index) {
-        return NilObject.nullToNil(truffleFrame.getValue(getBlockOrMethod().getStackSlot(index)));
+        return NilObject.nullToNil(FrameUtil.getObjectSafe(truffleFrame, getBlockOrMethod().getStackSlot(index)));
```

Comment:
Well here, you have to use `truffleFrame.getValue()` I believe because you didn't check `truffleFrame.isObject(slot)`.